### ISSUE TITLE
False positive in `no-thenable`

### ISSUE THREAD

**Issue by mcmxcdev** [sindresorhus/eslint-plugin-unicorn#1710]
I am using `eslint-plugin-unicorn` in a project with `yup` for form validation. `yup` has a syntax which includes `then()` like can be seen here: https://github.com/jquense/yup#schemadescribeoptions-resolveoptions-schemadescription

The example below shows where the lint check is triggering:
```js
validationSchema: Yup.object().shape(
        {
          addEntry: Yup.string()
            .nullable()
            .notRequired()
            .when('addEntry', {
              is: () => entryType === 'EMAIL',
              // eslint-disable-next-line unicorn/no-thenable
              then: (rule) => rule.email('Please enter a valid email.'),
              otherwise: (rule) =>
                rule.matches(
                  anyDomainRegex,
                  'Please enter a correct domain format.'
                ),
            }),
        },
        [
          ['addEntry', 'addEntry'],
        ]
      ),
```

You can see from the example where I had to disable eslint to silence the warning.

**Comment by fregante** [comment]
I think rules will always conflict with objects that are handled elsewhere. REST APIs and naming conventions are always problematic for example

**Comment by fisker** [comment]
You can disable this rule when your project uses `yup`, or use `eslint-disable`.

**Comment by samuelpucat** [comment]
I went around this issue by using .when() [function syntax](https://github.com/jquense/yup#schemawhenkeys-string--string-builder-object--values-any-schema--schema-schema). The rule is still turned on and 0 false positive issues with yup library.
